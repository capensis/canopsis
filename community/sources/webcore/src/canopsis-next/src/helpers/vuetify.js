import { VUETIFY_ANIMATION_DELAY } from '@/config';

/**
 * Wait a vuetify animation
 *
 * @return {Promise}
 */
export const waitVuetifyAnimation = () => new Promise(resolve => setTimeout(resolve, VUETIFY_ANIMATION_DELAY));
