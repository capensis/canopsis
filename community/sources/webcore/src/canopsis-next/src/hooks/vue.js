import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
