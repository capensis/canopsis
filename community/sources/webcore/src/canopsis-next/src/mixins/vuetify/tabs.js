export const vuetifyTabsMixin = {
  methods: {
    callTabsOnResizeMethod() {
      this.$refs?.tabs?.onResize?.();
    },
    callTabsUpdateTabsMethod() {
      this.$refs?.tabs?.updateTabsView?.();
    },
  },
};
