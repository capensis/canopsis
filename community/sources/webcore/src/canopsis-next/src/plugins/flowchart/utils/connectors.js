import { range } from 'lodash';

import uid from '@/helpers/uid';

/**
 * @typedef {Object} ConnectorPoint
 * @property {number} x
 * @property {number} y
 * @property {string} _id
 */

/**
 * Generate connector
 *
 * @param {number} x
 * @param {number} y
 * @returns {ConnectorPoint}
 */
export const generateConnectorPoint = (x, y) => ({
  x,
  y,
  _id: uid(),
});

/**
 * Generate connector points by rectangle
 *
 * @param {Rect} rect
 * @param {number} count
 * @returns {ConnectorPoint[]}
 */
export const generateConnectorsByRect = (rect, count) => {
  const { width, height, rx, ry, x, y } = rect;
  const sideConnectorsCount = Math.round(count / 4);
  const sideSpacesCount = sideConnectorsCount + 1;
  const startX = x + rx;
  const startY = x + ry;

  const widthConnectorOffset = (width - rx * 2) / (sideSpacesCount);
  const heightConnectorOffset = (height - ry * 2) / (sideSpacesCount);

  const sideConnectors = range(1, sideConnectorsCount + 1);

  const topConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y,
  ));

  const rightConnectors = sideConnectors.map(index => generateConnectorPoint(
    x + width,
    startY + heightConnectorOffset * index,
  ));

  const bottomConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y + height,
  ));

  const leftConnectors = sideConnectors.map(index => generateConnectorPoint(
    x,
    startY + heightConnectorOffset * index,
  ));

  const connectors = [
    ...topConnectors,
    ...rightConnectors,
    ...bottomConnectors,
    ...leftConnectors,
  ];

  if (!rx && !ry) {
    /* Angles connectors */
    connectors.push(
      generateConnectorPoint(x, y),
      generateConnectorPoint(x + width, y),
      generateConnectorPoint(x + width, y + height),
      generateConnectorPoint(x, y + height),
    );
  }

  return connectors;
};
