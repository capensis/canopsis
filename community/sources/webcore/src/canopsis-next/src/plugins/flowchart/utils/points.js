/**
 * Calculate points with ghosts points
 *
 * @param {Array} points
 * @returns {Array}
 */
export const getPointsWithGhosts = points => points.reduce((acc, point, index) => {
  const nextIndex = index + 1;
  const nextPoint = points[nextIndex];

  acc.push({ ...point, index });

  if (nextPoint) {
    acc.push({
      x: (point.x + nextPoint.x) / 2,
      y: (point.y + nextPoint.y) / 2,
      ghost: true,
      index: nextIndex,
    });
  }

  return acc;
}, []);
