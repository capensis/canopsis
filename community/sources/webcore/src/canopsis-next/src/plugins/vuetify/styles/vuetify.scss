@import "components/v-data-table";
@import "components/v-icon";
@import "components/v-tabs";
@import "components/v-text-field";
@import "components/v-window";
@import "components/v-card";
@import "components/v-toolbar";
