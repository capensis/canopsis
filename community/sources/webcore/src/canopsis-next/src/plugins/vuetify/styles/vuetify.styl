$default-font-size = 14px

rem(px) {
  return unit(px / $default-font-size, 'rem');
}

// Font sizes
$font-size-root = var(--v-font-size-root, $default-font-size)
$alert-font-size = $font-size-root

$list-tile-font-size = rem(16px)
$list-action-text-font-size = rem(12px)
$list-sub-title-font-size = rem(14px)
$list-item-dense-font-size = rem(13px)

$button-font-size = rem(14px)
$button-small-font-size = rem(13px)
$button-large-font-size = rem(15px)

$badge-font-size = rem(14px)

$breadcrumbs-item-font-size = rem(14px)
$breadcrumbs-item-large-font-size = rem(16px)
$chip-font-size = rem(13px)
$chip-small-font-size = rem(20px)

$subheader-height-font-size = rem(14px)
$subheader-dense-font-size = $list-item-dense-font-size

$time-picker-number-font-size = rem(16px)

$navigation-drawer-item-font-size = rem(14px)

$calendar-daily-weekday-font-size = rem(14px)
$calendar-daily-day-font-size = rem(40px)
$calendar-daily-interval-gutter-font-size = rem(10px)
$calendar-weekly-weekday-font-size = rem(14px)

$alert-icon-font-size = rem(24px)
$chip-icon-font-size = rem(20px)
$list-item-dense-icon-font-size = rem(22px)

@import '~vuetify/src/stylus/theme.styl'

// Backgrounds
// Light
$material-light.bg-color = var(--v-application-background-base, $material-light.background)
$material-light.background = var(--v-application-background-base, $material-light.background)
$material-light.cards = var(--v-background-base, $material-dark.cards)

$material-light.text.primary = var(--v-text-light-primary, $material-light.text.primary)
$material-light.text.secondary = var(--v-text-light-secondary, $material-light.text.secondary)
$material-light.text.disabled = var(--v-text-light-disabled, $material-light.text.disabled)

$material-light.icons.active = $material-light.text.secondary

$material-light.buttons.disabled = var(--v-button-light-disabled, $material-light.buttons.disabled)
$material-light.buttons.focused = var(--v-button-light-focused, $material-light.buttons.focused)

$material-light.chips.color = var(--v-text-light-primary, $material-light.chips.color)

$material-light.tabs.active = var(--v-text-light-primary, $material-light.chips.color)

$material-light.stepper.active = var(--v-stepper-active, $material-light.stepper.active)
$material-light.stepper.completed = var(--v-stepper-completed, $material-light.stepper.completed)
$material-light.stepper.hover = var(--v-stepper-hover, $material-light.stepper.hover)

// Dark
$material-dark.background = var(--v-application-background-base, $material-dark.background)
$material-dark.cards = var(--v-background-base, $material-dark.cards)

$material-dark.text.primary = var(--v-text-dark-primary, $material-dark.text.primary)
$material-dark.text.secondary = var(--v-text-dark-secondary, $material-dark.text.secondary)
$material-dark.text.disabled = var(--v-text-dark-disabled, $material-dark.text.disabled)

$material-dark.icons.active = $material-dark.text.secondary

$material-dark.buttons.disabled = var(--v-button-dark-disabled, $material-dark.buttons.disabled)
$material-dark.buttons.focused = var(--v-button-dark-focused, $material-dark.buttons.focused)

$material-dark.chips.color = var(--v-text-dark-disabled, $material-dark.chips.color)

$material-dark.tabs.active = var(--v-text-dark-primary, $material-dark.chips.color)

$material-dark.stepper.active = var(--v-stepper-active, $material-dark.stepper.active)
$material-dark.stepper.completed = var(--v-stepper-completed, $material-dark.stepper.completed)
$material-dark.stepper.hover = var(--v-stepper-hover, $material-dark.stepper.hover)

@import '~vuetify/src/stylus/settings/_variables.styl'

// Font sizes
$headings.h1.size = rem(112px)
$headings.h2.size = rem(56px)
$headings.h3.size = rem(45px)
$headings.h4.size = rem(34px)
$headings.h5.size = rem(24px)
$headings.h6.size = rem(20px)
$headings.subheading.size = rem(16px)
$headings.body-2.size = rem(14px)
$headings.body-1.size = rem(14px)
$headings.caption.size = rem(12px)

@import '~vuetify/src/stylus/main'
@import '~vuetify/src/stylus/components/_lists.styl'
@import '~vuetify/src/stylus/components/_navigation-drawer.styl'

@import './components/data-table.styl'
@import './components/steppers.styl'

table.v-table
  thead
    th
      font-size: rem(12px)

  tbody
    td
      font-size: rem(13px)

.v-tooltip
  &__content
    color: var(--v-text-dark-primary, #fff)
    font-size: rem(12px)

.v-input
  font-size: rem(16px)
