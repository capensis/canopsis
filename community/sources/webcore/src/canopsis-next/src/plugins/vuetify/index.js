import Vuetify from 'vuetify/lib';
import * as VuetifyComponents from 'vuetify/lib/components';
import * as VuetifyDirectives from 'vuetify/lib/directives';

import icons from './components/icons';
// import ClickOutside from './directives/click-outside';
// import VTabs from './components/v-tabs/v-tabs.vue';
// import VSelect from './components/v-select';
// import VTextarea from './components/v-textarea/v-textarea.vue';
// import VTooltip from './components/v-tooltip/v-tooltip.vue';
// import VMenu from './components/v-menu/v-menu.vue';
// import VListGroup from './components/v-list-group/v-list-group.vue';
// import VNavigationDrawer from './components/v-navigation-drawer/v-navigation-drawer.vue';
// import VExpansionPanelContent from './components/v-expansion-panel-content/v-expansion-panel-content.vue';
// import VDialog from './components/v-dialog/v-dialog.vue';
// import VCombobox from './components/v-combobox/v-combobox.vue';
// import VChipGroup from './components/v-chip-group/v-chip-group.vue';
// import VChip from './components/v-chip/v-chip.vue';
import VDataTable from './components/v-data-table/v-data-table.vue';
import VSpeedDial from './components/v-speed-dial/v-speed-dial.vue';

import './styles/vuetify.scss';

export const createVuetify = (Vue, options) => {
  Vue.use(Vuetify, {
    components: {
      ...VuetifyComponents,

      VSpeedDial,
      VDataTable,
    },

    directives: {
      ...VuetifyDirectives,
    },
  });

  // Vue.component('v-chip', VChip);
  // Vue.component('v-chip-group', VChipGroup);
  // Vue.component('v-combobox', VCombobox);
  // Vue.component('v-dialog', VDialog);
  // Vue.component('v-expansion-panel-content', VExpansionPanelContent);
  // Vue.component('v-navigation-drawer', VNavigationDrawer);
  // Vue.component('v-list-group', VListGroup);
  // Vue.component('v-menu', VMenu);
  // Vue.component('v-tooltip', VTooltip);
  // Vue.component('v-data-table', VDataTable);
  // Vue.component('v-textarea', VTextarea);
  // Vue.component('v-select', VSelect);
  // Vue.component('v-tabs', VTabs);

  // Vue.directive('click-outside', ClickOutside);

  return new Vuetify({
    icons: {
      iconfont: 'md',
      values: icons,
    },
    theme: {
      options: {
        customProperties: true,
      },
      ...options.theme,
    },
  });
};
