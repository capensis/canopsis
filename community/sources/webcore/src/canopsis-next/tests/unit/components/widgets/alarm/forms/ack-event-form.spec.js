import Faker from 'faker';
import flushPromises from 'flush-promises';

import { generateShallowRenderer, generateRenderer } from '@unit/utils/vue';
import { createCheckboxInputStub } from '@unit/stubs/input';
import { createActivatorElementStub } from '@unit/stubs/vuetify';

import AckEventForm from '@/components/widgets/alarm/forms/ack-event-form.vue';

const stubs = {
  'c-description-field': true,
  'v-checkbox': createCheckboxInputStub('v-checkbox'),
  'v-tooltip': createActivatorElementStub('v-tooltip'),
};

const snapshotStubs = {
  'c-description-field': true,
};

const selectDescriptionField = wrapper => wrapper.find('c-description-field-stub');
const selectCheckboxField = wrapper => wrapper.find('.v-checkbox');

describe('ack-event-form', () => {
  const factory = generateShallowRenderer(AckEventForm, {
    stubs,
    attachTo: document.body,
  });
  const snapshotFactory = generateRenderer(AckEventForm, {
    stubs: snapshotStubs,
    attachTo: document.body,
  });

  test('Output changed after trigger description field', () => {
    const form = {
      output: Faker.datatype.string(),
      ack_resources: Faker.datatype.boolean(),
    };
    const wrapper = factory({
      propsData: {
        form,
      },
    });

    const output = Faker.datatype.string();

    const descriptionField = selectDescriptionField(wrapper);

    descriptionField.vm.$emit('input', output);

    expect(wrapper).toEmit('input', { ...form, output });
  });

  test('Ack resources changed after trigger checkbox field', async () => {
    const form = {
      output: Faker.datatype.string(),
      ack_resources: Faker.datatype.boolean(),
    };
    const wrapper = factory({
      propsData: {
        form,
      },
    });

    await flushPromises();

    const ackResources = !form.ack_resources;

    selectCheckboxField(wrapper).vm.$emit('change', ackResources);

    expect(wrapper).toEmit('input', { ...form, ack_resources: ackResources });
  });

  test('Renders `ack-event-form` with default props', () => {
    const wrapper = snapshotFactory();

    expect(wrapper.element).toMatchSnapshot();
  });

  test('Renders `ack-event-form` with custom props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        form: {
          output: 'output',
          ack_resources: true,
        },
        isNoteRequired: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
