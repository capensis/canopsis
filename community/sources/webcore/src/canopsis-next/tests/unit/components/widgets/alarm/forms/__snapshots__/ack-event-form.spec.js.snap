// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ack-event-form Renders \`ack-event-form\` with custom props 1`] = `
<div
  class="layout column"
>
  <div
    class="v-input v-text-field v-input--is-label-active v-input--is-dirty theme--light"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <label
            aria-hidden="true"
            class="v-label v-label--active theme--light"
            style="left: 0px; position: absolute;"
          >
            Ticket number
          </label>
          <input
            aria-label="Ticket number"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <div
            class="v-messages__wrapper"
          />
        </div>
      </div>
    </div>
  </div>
  <c-description-field-stub
    label="Note"
    required="true"
    value="output"
  />
  <div
    class="layout row"
  >
    <span
      class="v-tooltip v-tooltip--top"
    >
      <transition-stub
        name="slide-y-reverse-transition"
      />
      <div
        class="v-input v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty theme--light"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-input--selection-controls__input"
            >
              <input
                aria-checked="true"
                aria-label="Ack resources"
                role="checkbox"
                type="checkbox"
                value=""
              />
              <div
                class="v-input--selection-controls__ripple primary--text"
              />
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                check_box
              </i>
            </div>
            <label
              aria-hidden="true"
              class="v-label theme--light"
              style="left: 0px; position: relative;"
            >
              Ack resources
            </label>
          </div>
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`ack-event-form Renders \`ack-event-form\` with default props 1`] = `
<div
  class="layout column"
>
  <div
    class="v-input v-text-field theme--light"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <label
            aria-hidden="true"
            class="v-label theme--light"
            style="left: 0px; position: absolute;"
          >
            Ticket number
          </label>
          <input
            aria-label="Ticket number"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <div
            class="v-messages__wrapper"
          />
        </div>
      </div>
    </div>
  </div>
  <c-description-field-stub
    label="Note"
  />
  <div
    class="layout row"
  >
    <span
      class="v-tooltip v-tooltip--top"
    >
      <transition-stub
        name="slide-y-reverse-transition"
      />
      <div
        class="v-input v-input--selection-controls v-input--checkbox theme--light"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-input--selection-controls__input"
            >
              <input
                aria-checked="false"
                aria-label="Ack resources"
                role="checkbox"
                type="checkbox"
                value=""
              />
              <div
                class="v-input--selection-controls__ripple"
              />
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                check_box_outline_blank
              </i>
            </div>
            <label
              aria-hidden="true"
              class="v-label theme--light"
              style="left: 0px; position: relative;"
            >
              Ack resources
            </label>
          </div>
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;
